### PR TITLE
[INLONG-7391][Sort] Support CSV format and dirty data collecting for StarRocks connector all db migrating

### LINKED ISSUE #7391: [Improve][Sort] Support CSV format and dirty data collecting for StarRocks connector
### Description

When migrating the whole db, the records are composed with JSON format to submit to StarRocks Server. JSON data may cause StarRocks Server high CPU loads. Data with CSV format will reduce CPU loads.

Records will be lost if an exception threw by StarRocks Server.  We may upload the records to S3 bucket as dirty data. This is useful when auditing and checking.

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liaorui**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #7391

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

When migrating the whole db, the records are composed with JSON format to submit to StarRocks Server. JSON data may cause StarRocks Server high CPU loads. Data with CSV format will reduce CPU loads.

Records will be lost if an exception threw by StarRocks Server. We may upload the records to S3 bucket as dirty data. This is useful when auditing and checking.

### Modifications

*Describe the modifications you've done.*

1. DirtySinkHelper in base module has new `regexReplace` method. It can replace the meta field with real data, such as DIRTY_TYPE、DIRTY_MESSAGE and SYSTEM_TIME.
2. SinkTableMetricData in base module has new `outputDirtyMetrics` method. 
3. SinkBufferEntity overwrites `StarRocksSinkBufferEntity`. Each batches carry dirty meta data and columns header. When composing the CSV data, columns will be used as http header. Dirty meta data will be used to upload dirty data when the server threw an exception.
4. StarRocksSinkManager has new `writeRecords` method. It will compose data with CSV or JSON format by `sink.properties.format` configuration.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-connectors/base/src/test/java/org/apache/inlong/sort/base/dirty/RegexReplaceTest.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public void testRegexReplacement() throws IOException {
         identifier[0] = "yizhouyang";
         identifier[1] = "table2";
         String pattern = "${database}-${table}-${DIRTY_MESSAGE}";
-        String answer = DirtySinkHelper.regexReplace(pattern, DirtyType.BATCH_LOAD_ERROR, "mock message", identifier);
+        String answer = DirtySinkHelper.regexReplace(pattern, DirtyType.BATCH_LOAD_ERROR, "mock message");
```
Comment (gong) [inline inlong-sort/sort-connectors/base/src/test/java/org/apache/inlong/sort/base/dirty/RegexReplaceTest.java:35]: Please fix this UT error

Comment (gong) [reply inlong-sort/sort-connectors/base/src/test/java/org/apache/inlong/sort/base/dirty/RegexReplaceTest.java:35]: @Yizhou-Yang Please check this UT modication

Comment (liaorui) [reply inlong-sort/sort-connectors/base/src/test/java/org/apache/inlong/sort/base/dirty/RegexReplaceTest.java]: Fixed.

**Review by gong — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

**Pull request merged**